app/models/comment.php

```diff
@@ -47,6 +84,73 @@ public static function getAll($thread_id, $offset, $limit)
         return $comments;
     }
 
+    public static function getAllByUser(User $user)
+    {
+        $db     = DB::conn();
+        $rows   = $db->rows(
+            sprintf('SELECT * FROM %s WHERE user_id=? ORDER BY id DESC', self::TABLE_NAME),
+            array($user->id)
+        );
+
+        $comments = array();
+
+        foreach ($rows as $row) {
+            $comments[] = new self($row);
+        }
+
+        return $comments;
+    }
+
+
+    public static function get($id)
+    {
+        $db     = DB::conn();
+        $row    = $db->row(sprintf("SELECT * FROM %s WHERE id=?", self::TABLE_NAME), array($id));
+
+        return $row ? new self($row) : false;
+    }
+
+    public static function getOrFail($id)
+    {
+        $comment = self::get($id);
+
+        if ($comment) {
+            return $comment;
+        } else {
+            throw new RecordNotFoundException();
+        }
+    }
+
+    public static function getFirstInThread(Thread $thread)
+    {
+        $db     = DB::conn();
+        $row    = $db->row(sprintf('SELECT * FROM %s WHERE thread_id=? LIMIT 0, 1', self::TABLE_NAME), array($thread->id));
+
+        return $row ? new self($row) : false;
+    }
+
+    public static function getLastInThread(Thread $thread)
+    {
+        $db     = DB::conn();
+        $row    = $db->row(sprintf('SELECT * FROM %s WHERE thread_id=? ORDER BY id DESC LIMIT 0, 1', self::TABLE_NAME), array($thread->id));
+
+        return $row ? new self($row) : false;
+    }
```

Comment:
This is just a suggestion po. If you do not have the time to do this, then it's fine po :smile: